lib/SimpleSAML/Module.php

```diff
@@ -265,11 +265,21 @@ public static function process(Request $request = null)
         }
 
         $response = new BinaryFileResponse($path);
-        $response->setCache(['public' => true, 'max_age' => 86400]);
-        $response->setExpires(new \DateTime(gmdate('D, j M Y H:i:s \G\M\T', time() + 10 * 60)));
-        $response->setLastModified(new \DateTime(gmdate('D, j M Y H:i:s \G\M\T', filemtime($path))));
+        $cacheSettings = $config->getArray('module.php.cache', ['public' => true, 'max_age' => 86400]);
+        $response->setCache($cacheSettings);
+        $cacheExpiration = $config->getValue('module.php.expires', 10 * 60);
+        $expires = $cacheExpiration === null ? null : new \DateTime('@' . (time() + $cacheExpiration));
+        $response->setExpires($expires);
+        $cacheEtag = $config->getBoolean('module.php.etag', false);
+        if ($cacheEtag) {
+            $response->setAutoEtag();
```

Comment:
This is a nice catch. It's definitely much more convenient to leave Symfony generate the `ETag` for us, so I think this is a good idea. In this case, we should probably also process the `etag` config option prior to the call to `setCache()`.

Reply:
What do you mean "process the `etag` config option prior to the call to `setCache()`?" I think we can move these calls around freely since they are nothing but setters.